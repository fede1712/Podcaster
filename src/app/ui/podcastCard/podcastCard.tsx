import { image } from "@/app/types";
import Image from "next/image";

export const PodcastCard = ({ src, title, author }: { src: image; title: string; author: string }) => {
  return (
    <div className="border rounded-md p-4 shadow-xl">
      <div className="flex justify-center relative bottom-14">
        <Image alt="cover" src={src.label} width={100} height={100} className="rounded-full" />
      </div>
      <section className="flex justify-center">
        <h2 className="text-center">{title.toLocaleUpperCase()}</h2>
      </section>
      <section className="flex justify-center">
        <p className="font-light text-sm text-center">Author: {author}</p>
      </section>
    </div>
  );
};
