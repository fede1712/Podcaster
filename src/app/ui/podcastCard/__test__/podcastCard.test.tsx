import React from "react";
import { render, screen } from "@testing-library/react";
import { PodcastCard } from "../podcastCard";

describe("Renders PodcastCard correctly", () => {
  test("should render PodcastCard", () => {
    render(<PodcastCard src={{ label: "http://example.com/artwork.jpg" }} title="Podcast title" author="Podcast author" podcastId="1" />);

    const titleElement = screen.queryByText("PODCAST TITLE");
    expect(titleElement).not.toBeNull();

    const authorElement = screen.queryByText("Author: Podcast author");
    expect(authorElement).not.toBeNull();

    const { asFragment } = render(
      <PodcastCard src={{ label: "http://example.com/artwork.jpg" }} title="Podcast title" author="Podcast author" podcastId="1" />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
