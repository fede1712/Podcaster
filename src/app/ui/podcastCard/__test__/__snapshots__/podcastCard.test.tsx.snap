// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders PodcastCard correctly should render PodcastCard 1`] = `
<DocumentFragment>
  <a
    href="podcast/1"
  >
    <div
      class="border rounded-md p-4 shadow-xl"
    >
      <div
        class="flex justify-center relative bottom-14"
      >
        <img
          alt="cover"
          class="rounded-full"
          data-nimg="1"
          decoding="async"
          height="100"
          loading="lazy"
          src="/_next/image?url=http%3A%2F%2Fexample.com%2Fartwork.jpg&w=256&q=75"
          srcset="/_next/image?url=http%3A%2F%2Fexample.com%2Fartwork.jpg&w=128&q=75 1x, /_next/image?url=http%3A%2F%2Fexample.com%2Fartwork.jpg&w=256&q=75 2x"
          style="color: transparent;"
          width="100"
        />
      </div>
      <section
        class="flex justify-center"
      >
        <h2
          class="text-center"
        >
          PODCAST TITLE
        </h2>
      </section>
      <section
        class="flex justify-center"
      >
        <p
          class="font-light text-sm text-center"
        >
          Author: Podcast author
        </p>
      </section>
    </div>
  </a>
</DocumentFragment>
`;
